src/main/java/teammates/ui/controller/InstructorEditInstructorFeedbackPageAction.java

```diff
@@ -0,0 +1,105 @@
+        String courseId = getRequestParamValue(Const.ParamsNames.COURSE_ID); 
+        String feedbackSessionName = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_NAME);
+        String moderatedEntityIdentifier = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_MODERATED_STUDENT);
+        String moderatedQuestionNumber = getRequestParamValue("moderatedquestion");
+
+        Assumption.assertNotNull(String.format(Const.StatusMessages.NULL_POST_PARAMETER_MESSAGE, 
+                                               Const.ParamsNames.COURSE_ID), 
+                                 courseId);
+        Assumption.assertNotNull(String.format(Const.StatusMessages.NULL_POST_PARAMETER_MESSAGE, 
+                                               Const.ParamsNames.FEEDBACK_SESSION_NAME), 
+                                 feedbackSessionName);
+        Assumption.assertNotNull(String.format(Const.StatusMessages.NULL_POST_PARAMETER_MESSAGE, 
+                                               Const.ParamsNames.FEEDBACK_SESSION_MODERATED_STUDENT), 
+                                 moderatedEntityIdentifier);
+
+        new GateKeeper().verifyAccessible(
+                logic.getInstructorForGoogleId(courseId, account.googleId), 
+                logic.getFeedbackSession(feedbackSessionName, courseId),
+                false, Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_SESSION);
+        
+        InstructorAttributes instructorUnderModeration = logic.getInstructorForEmail(courseId, moderatedEntityIdentifier);
+
+        // If the instructor doesn't exist
+        if (instructorUnderModeration == null) {
+            throw new EntityDoesNotExistException("Instructor Email "
+                    + moderatedEntityIdentifier + " does not exist in " + courseId
+                    + ".");
+        }
+
+        FeedbackSubmissionEditPageData data = new FeedbackSubmissionEditPageData(account, student);
+
+        data.bundle = logic.getFeedbackSessionQuestionsBundleForInstructor(
+                feedbackSessionName, 
+                courseId, 
+                instructorUnderModeration.email);
+
+        Assumption.assertNotNull(data.bundle);
+        
+        data.setSessionOpenForSubmission(true);
+        data.setModeration(true);
+        data.setHeaderHidden(true);
+        data.setPreviewInstructor(instructorUnderModeration);
+        data.setSubmitAction(Const.ActionURIs.INSTRUCTOR_EDIT_INSTRUCTOR_FEEDBACK_SAVE);
+
+        if (moderatedQuestionNumber != null) {
+            data.setModeratedQuestion(moderatedQuestionNumber);
+        }
+
+        hideQuestionsWithAnonymousResponses(data.bundle);
+
+        statusToAdmin = "Moderating feedback session for instructor (" + instructorUnderModeration.email + ")<br>" +
+                        "Session Name: " + feedbackSessionName + "<br>" +
+                        "Course ID: " + courseId;
+        
+        data.init("", "", courseId);
```

Comment:
This does not make sense just by looking at the function call... we should refactor this.